src/rust/lib/graph-editor/src/component/visualization/class.rs

```diff
@@ -154,3 +176,126 @@ impl Visualization {
         self.state.renderer.set_size(size)
     }
 }
+
+
+
+// ===========================
+// === Visualization Class ===
+// ===========================
+
+/// Result of the attempt to instantiate a `Visualization` from a `Class`.
+pub type InstantiationResult = Result<Visualization,Box<dyn Error>>;
+
+/// Specifies a trait that allows the instantiation of `Visualizations`.
+///
+/// The `Class` provides a way to implement structs that allow the instantiation of specific
+/// visualizations, while already providing general information that doesn't require an
+/// instantiated visualization, for example, the name or input type of the visualization.
+///
+/// There are two example implementations: The `JsSourceClass`, which is based on a JS snippet to
+/// instantiate `JsRenderer`, and the fairly generic `NativeConstructorClass`, that only requires
+/// a function that can create a InstantiationResult. The later can be used as a thin wrapper around
+/// the constructor methods of native visualizations.
+///
+/// Example
+/// --------
+/// ```no_run
+/// use graph_editor::component::visualization;
+/// use graph_editor::component::visualization::Visualization;
+/// use graph_editor::component::visualization::renderer::example::native::BubbleChart;
+/// use ensogl::display::Scene;
+/// use std::rc::Rc;
+///
+/// // Create a `visualization::Class` from a JS source code snippet.
+/// let js_source_class = visualization::JsSourceClass::from_js_source_raw(r#"
+///
+///    class BubbleVisualization {
+///         static inputTypes = ["[[float;3]]"]
```

Comment:
These should be Enso types, so if you want to have a list of 3 floats, you write `[Float,Float,Float]` instead of `[float;3]` :) If you want to have a List of unknown length then you write `List Float`.